apps/etterlatte-saksbehandling-ui/client/src/components/behandling/virkningstidspunkt/Virkningstidspunkt.tsx

```diff
@@ -5,74 +5,90 @@ import { oppdaterBehandlingsstatus, oppdaterVirkningstidspunkt } from '~store/re
-import { KildePdl } from '~shared/types/kilde'
-import { hentMinimumsVirkningstidspunkt } from '~components/behandling/soeknadsoversikt/soeknadoversikt/virkningstidspunkt/utils'
+import { SoeknadsoversiktTextArea } from '~components/behandling/soeknadsoversikt/SoeknadsoversiktTextArea'
+import { hentMinimumsVirkningstidspunkt } from '~components/behandling/virkningstidspunkt/utils'
 import { UseMonthPickerOptions } from '@navikt/ds-react/esm/date/hooks/useMonthPicker'
 
-interface Props {
-  behandlingId: string
-  redigerbar: boolean
-  virkningstidspunkt: Virkningstidspunkt | null
-  avdoedDoedsdato: string | undefined
-  avdoedDoedsdatoKilde: KildePdl | undefined
-  soeknadMottattDato: string
-  hjemmler: Hjemmel[]
-  beskrivelse: string
-  children?: { info: React.ReactNode }
-}
-
 export interface Hjemmel {
   lenke: string
   tittel: string
 }
 
-const Virkningstidspunkt = (props: Props) => {
+const Virkningstidspunkt = (props: {
+  behandling: IDetaljertBehandling
+  redigerbar: boolean
+  hjemler: Hjemmel[]
+  beskrivelse: string
+  children?: { info: React.ReactNode }
+  erBosattUtland: boolean
+}) => {
+  const { behandling, erBosattUtland } = props
   const dispatch = useAppDispatch()
 
-  const [vurdert, setVurdert] = useState(props.virkningstidspunkt !== null)
-  const [formData, setFormData] = useState<Date | null>(
-    props.virkningstidspunkt ? new Date(props.virkningstidspunkt.dato) : null
+  const [vurdert, setVurdert] = useState(behandling.virkningstidspunkt !== null)
+  const [virkningstidspunkt, setVirkningstidspunkt] = useState<Date | null>(
+    behandling.virkningstidspunkt ? new Date(behandling.virkningstidspunkt.dato) : null
   )
   const [, fastsettVirkningstidspunktRequest, resetToInitial] = useApiCall(fastsettVirkningstidspunkt)
-  const [begrunnelse, setBegrunnelse] = useState<string>(props.virkningstidspunkt?.begrunnelse ?? '')
+
+  const [begrunnelse, setBegrunnelse] = useState<string>(behandling.virkningstidspunkt?.begrunnelse ?? '')
   const [errorTekst, setErrorTekst] = useState<string>('')
 
+  const [kravdato, setKravdato] = useState<Date | undefined>(undefined)
+
+  const { monthpickerProps: monthpickerPropsKravdato, inputProps: inputPropsKravdato } = useMonthpicker({
+    fromDate: subYears(new Date(), 20),
+    toDate: addMonths(new Date(), 10),
+    inputFormat: 'dd.MM.yyyy',
+    onMonthChange: (date: Date | undefined) => setKravdato(date),
+    defaultSelected: kravdato ?? undefined,
+  })
+
+  const avdoedDoedsdato = behandling.familieforhold?.avdoede?.opplysning?.doedsdato
+
   const { monthpickerProps, inputProps } = useMonthpicker({
-    fromDate: hentMinimumsVirkningstidspunkt(props.avdoedDoedsdato, props.soeknadMottattDato),
+    fromDate: hentMinimumsVirkningstidspunkt(
+      avdoedDoedsdato,
+      erBosattUtland ? subYears(new Date(), 20) : new Date(behandling.soeknadMottattDato)
```

Comment:
Spørsmålet er man egentlig vil ha en beregning på datoen her for bu. Greia er at selvom mans ender inn riktig dato til `hentMinimumsVirkningstidspunkt` så oppdaterer ikke usemonthpicker seg, mao datoene blir ikke synlige etter første render. Som krever en hel del omskrivning så vet ikke om det er nødvendig. Det er uansett på plass backend validering.